app/controllers/api/v7/works_controller.rb

```diff
@@ -138,6 +158,14 @@ def get_ids(params)
       collection = Work.where(publisher_id: publisher.id)
     elsif params[:contributor_id] && contributor = Contributor.where(pid: params[:contributor_id]).first
       collection = Work.joins(:contributions).where("contributions.contributor_id = ?", contributor.id)
+    elsif params[:id]
+      id_hash = get_id_hash(params[:id])
+      if id_hash.respond_to?("key")
```

Comment:
Great, thank you very much! Sorry for being picky, but I want to prevent as much potentially hard to understand code entering the codebase.